src/ros_filter.cpp

```diff
@@ -1065,7 +1066,11 @@ namespace RobotLocalization
           relative = false;
         }
 
-        std::string odomTopic;
+        // Consider odometry transformation from the child_frame_id instead of the base_link_frame_id
```

Comment:
If we only apply the new behavior to `relative` and `differential` mode, you're right. In this case, I will change it as suggested. It would help the user find inconsistencies in his configuration.

If we apply the new behavior to the *normal* mode as well, I rather propose to rename `diff_rel_use_child_frame` to `pose_uses_child_frame`. Then, setting the new parameter to `true` even makes sense without setting `relative` or `differential`.

Reply:
Done.